cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/PersistedMessageCallback.java

```diff
@@ -117,13 +119,10 @@ public void remove() {
     }
 
     private void recordLastOffset() {
+      messageCount.incrementAndGet();
```

Comment:
Moving `messageCount.incrementAndGet()` to its original place would make code clearer.